### ISSUE TITLE
Conflicting configuration for Backup VM policies after upgrade to Pulumi.Azure 5.45.0 package

### ISSUE THREAD

**Issue by jrgcubano** [pulumi/pulumi-azure#1318]
### What happened?

 I have upgraded some of our C# IaC projects to the latest Pulumi (Pulumi: 3.55.1 and Pulumi.Azure: 5.45.0) and when trying to do 'pulumi up' I am getting the following conflicts with the vm backup policies:

```
Previewing update (appyway/qa)

View in Browser (Ctrl+O): https://app.pulumi.com/appyway/BugReport.BackupPolicy/qa/previews/803f9387-0ba4-4496-bef2-5b7f30762ef2

     Type                      Name                                  Plan     Info
     pulumi:pulumi:Stack       BugReport.BackupPolicy-qa
     └─ azure:backup:PolicyVM  rsvpol-qa-bugrbak-eun-vmbackuppolicy           6 errors

Diagnostics:
  azure:backup:PolicyVM (rsvpol-qa-bugrbak-eun-vmbackuppolicy):
    error: azure:backup/policyVM:PolicyVM resource 'rsvpol-qa-bugrbak-eun-vmbackuppolicy' has a problem: Conflicting configuration arguments: "retention_monthly.0.weeks": conflicts with retention_monthly.0.include_last_days. Examine values at 'rsvpol-qa-bugrbak-eun-vmbackuppolicy.retentionMonthly.0.weeks'.
    error: azure:backup/policyVM:PolicyVM resource 'rsvpol-qa-bugrbak-eun-vmbackuppolicy' has a problem: Conflicting configuration arguments: "retention_monthly.0.weekdays": conflicts with retention_monthly.0.include_last_days. Examine values at 'rsvpol-qa-bugrbak-eun-vmbackuppolicy.retentionMonthly.0.weekdays'.
    error: azure:backup/policyVM:PolicyVM resource 'rsvpol-qa-bugrbak-eun-vmbackuppolicy' has a problem: Conflicting configuration arguments: "retention_monthly.0.include_last_days": conflicts with retention_monthly.0.weeks. Examine values at 'rsvpol-qa-bugrbak-eun-vmbackuppolicy.retentionMonthly.0.includeLastDays'.
    error: azure:backup/policyVM:PolicyVM resource 'rsvpol-qa-bugrbak-eun-vmbackuppolicy' has a problem: Conflicting configuration arguments: "retention_yearly.0.include_last_days": conflicts with retention_yearly.0.weeks. Examine values at 'rsvpol-qa-bugrbak-eun-vmbackuppolicy.retentionYearly.0.includeLastDays'.
    error: azure:backup/policyVM:PolicyVM resource 'rsvpol-qa-bugrbak-eun-vmbackuppolicy' has a problem: Conflicting configuration arguments: "retention_yearly.0.weeks": conflicts with retention_yearly.0.include_last_days. Examine values at 'rsvpol-qa-bugrbak-eun-vmbackuppolicy.retentionYearly.0.weeks'.
    error: azure:backup/policyVM:PolicyVM resource 'rsvpol-qa-bugrbak-eun-vmbackuppolicy' has a problem: Conflicting configuration arguments: "retention_yearly.0.weekdays": conflicts with retention_yearly.0.include_last_days. Examine values at 'rsvpol-qa-bugrbak-eun-vmbackuppolicy.retentionYearly.0.weekdays'.
```

### Expected Behavior

Have no conflicts after pulumi up.

### Steps to reproduce

- Example repository: https://github.com/jrgcubano/BugReports.BackupPolicy

### Output of `pulumi about`

```console
BugReport.BackupPolicy git:(master): pulumi about

running 'dotnet build -nologo .'
  Determining projects to restore...

  All projects are up-to-date for restore.

  BugReport.BackupPolicy -> C:\source\bugreports\pulumi\BugReports.BackupPolicy\BugReport.BackupPolicy\bin\Debug\net6.0\BugReport.BackupPolicy.dll



Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.45

'dotnet build -nologo .' completed successfully
CLI
Version      3.76.0
Go Version   go1.20.6
Go Compiler  gc

Plugins
NAME    VERSION
azure   5.45.0
dotnet  unknown

Host
OS       Microsoft Windows 11 Pro
Version  10.0.22621 Build 22621
Arch     x86_64

This project is written in dotnet: executable='C:\Program Files\dotnet\dotnet.exe' version='6.0.412'

Current Stack: appyway/BugReport.BackupPolicy/qa

TYPE                                    URN
pulumi:pulumi:Stack                     urn:pulumi:qa::BugReport.BackupPolicy::pulumi:pulumi:Stack::BugReport.BackupPolicy-qa
pulumi:providers:azure                  urn:pulumi:qa::BugReport.BackupPolicy::pulumi:providers:azure::default_5_43_0
azure:core/resourceGroup:ResourceGroup  urn:pulumi:qa::BugReport.BackupPolicy::azure:core/resourceGroup:ResourceGroup::rg-qa-bugrbak-eun
azure:recoveryservices/vault:Vault      urn:pulumi:qa::BugReport.BackupPolicy::azure:core/resourceGroup:ResourceGroup$azure:recoveryservices/vault:Vault::rsv-qa-bugrbak-eun
azure:backup/policyVM:PolicyVM          urn:pulumi:qa::BugReport.BackupPolicy::azure:backup/policyVM:PolicyVM::rsvpol-qa-bugrbak-eun-vmbackuppolicy


Found no pending operations associated with appyway/qa

Backend
Name           pulumi.com
URL            https://app.pulumi.com/Administrator
User           Administrator
Organizations  Administrator, appyway

Dependencies:
NAME          VERSION
Pulumi.Azure  5.45.0

Pulumi locates its logs in C:\Users\jrogalan\AppData\Local\Temp by default
```

### Additional context

Looking at the code in https://github.com/pulumi/pulumi-azure/commits/master it looks like the IncludeLastDays property was added in version [5.44.0](https://github.com/pulumi/pulumi-azure/releases/tag/sdk%2Fv5.44.0) => https://github.com/pulumi/pulumi-azure/commit/83aed27994d1d0d786dcafe20fc89f9138ea5389

![image](https://github.com/pulumi/pulumi-azure/assets/3245617/87a4d99b-1bab-456f-a8e0-134fbe5dda53)

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by rquitales** [comment]
@jrgcubano Thanks for reporting this issue and apologies you're experiencing this.

I believe this is an instance of https://github.com/pulumi/pulumi-terraform-bridge/issues/577. In the meantime, I might be able to fix this by adding a patch to our forked upstream provider. I'll post updates here once I have something working!

**Comment by jrgcubano** [comment]
@rquitales Thank you! Let me know if you need any more information.

**Comment by jrgcubano** [comment]
Hello @rquitales. Any news?

**Comment by rquitales** [comment]
@jrgcubano Sorry for the radio silence and thanks for the nudge - I've been out of office with personal commitments and this fell off my radar, will take a look now!

**Comment by rquitales** [comment]
@jrgcubano I've just released [v5.48.1](https://github.com/pulumi/pulumi-azure/releases/tag/v5.48.1). Could you test this new version and see if it resolves your issue? Thanks!
